### PR TITLE
#8 update urls

### PR DESCRIPTION
**Pull request by cdmo**
this affects the issues #8, #98 and will feed https://github.com/psu-libraries/psulib_blacklight/issues/226

we will likely need to update this based on further details about catalog data sooner than later

Connected to #98

### CODE REVIEW

**Comment by cdmo** [comment]
@banukutlu I officially release the WIP status of this PR and submit to your review. Godspeed.

File: spec/lib/traject/macros/custom_spec.rb [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  end
+  let(:result) { @indexer.map_record(MARC::Record.new_from_hash('fields' => [url_856], 'leader' => leader)) }
+
+  before(:all) do
+    c = './lib/traject/psulib_config.rb'
+    @indexer = Traject::Indexer.new
+    @indexer.load_config_file(c)
+  end
+
+  describe 'A record with a fulltext link because of indicator 2 is 0' do
```
Comment (banukutlu) [inline spec/lib/traject/macros/custom_spec.rb]: @cdmo this is minor: the description says "indicator 2 is 0" but the mocked field seems to have ind2 as 1 🤔

Comment (cdmo) [reply spec/lib/traject/macros/custom_spec.rb]: 🤦‍♂️

File: spec/lib/traject/macros/custom_spec.rb [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    let(:url_856_3a) do
+      { '856' => { 'ind1' => '0', 'ind2' => '2', 'subfields' => [{ 'u' => 'http://library.columbia.edu/content/library'\
+                                                                   'web/indiv/ccoh/our_work/how_to_use_the_archives.ht'\
+                                                                   'ml' },
+                                                                 { '3' => 'Carrots peas' }] } }
+    end
+    let(:result_3a) { @indexer.map_record(MARC::Record.new_from_hash('fields' => [url_856_3a], 'leader' => leader)) }
+
+    it 'produces a supplemental link' do
+      expect(result_3a['full_links_struct']).to match ['{"text":"library.columbia.edu","url":"http://library.columbia.'\
```
Comment (banukutlu) [inline spec/lib/traject/macros/custom_spec.rb]: @cdmo in this case, does it produce a supplemental or a fulltext link? or maybe just the description needs to be changed here?

Comment (cdmo) [reply spec/lib/traject/macros/custom_spec.rb]: full. oops.

**Pull request merged**